assets/javascripts/modules/checkout/promoCode.js

```diff
@@ -22,24 +23,79 @@ define([
         bindExtraKeyListener.alreadyBound = true;
     }
 
-    function displayError(message) {
-        message = message || 'Invalid promo code, please try again.';
+    function unapplyPromotionToRatePlans() {
+        $ratePlanFields.each(function(el) {
+            var $el = $(el),
+                currency = $el.data('currency'),
+                $label = $('#label-for-' + $el.val() + '-' + currency),
+                newDisplayPrice = $el.data('option-mirror-label-default');
+
+            $label.html(newDisplayPrice);
+            $el.attr('data-option-mirror-label', newDisplayPrice);
+            if ($el.attr('checked')) {
+                bean.fire(el, 'change');
+            }
+        });
+    }
+
+    function formatMoney(currencyIdentifier, amount) {
+        // rounds any remainder up tp 1p so we always quote a higher price than the user might eventually get charged.
+        var price2dp = Math.ceil(amount * 100) / 100,
+            priceNoDot00 = price2dp.toFixed(2).replace(/\.00$/, '');
+
+        return currencyIdentifier + priceNoDot00;
+    }
+
+    function applyPromotionToRatePlans(promotion) {
+        if (promotion.promotionType.amount > 0) {
```

Comment:
Agreed, and it should be linked in with the mirror-label thing too. Lodash templates is a good idea as it won't be a new full framework. I'll review what you guys have done in membership and we can bundle that change in to the limited-period-promotion work. I've captured this in comments for https://trello.com/c/3BiFWPkJ/385-data-requirements-for-offer-for-a-discounted-period-renewal-comms